lib/router/with-router.js

```diff
@@ -0,0 +1,34 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import hoistStatics from 'hoist-non-react-statics'
+
+export default function withRouter (options = {}) {
+  const { routerPropName = 'router' } = options
+
+  return function withRouterComponent (ComposedComponent) {
+    const displayName = ComposedComponent.displayName || ComposedComponent.name || 'Unknown'
```

Comment:
I don't know why he wants to merge this into `master` and not into `v3-beta`. 
I am not part of the official dev team.

Nevertheless if @arunoda allows you to change to branch `v3-beta` here is how it works: https://github.com/blog/2224-change-the-base-branch-of-a-pull-request. Otherwise we can update this issue in another PR.